Networking fixes

LeeVangraefschepe opened the pull request:
- Only the basepacket function got called when setting data for a packet.
- Client no longer needs ticks. The receive call is blocking and doesnt spam the thread.
- Changed error to warning when connection

LeeVangraefschepe committed:
Fixed a issue with set data in base packet

Only the basepacket function got called when setting data for the packet.

LeeVangraefschepe committed:
Added UDP reveive support

Client no longer needs ticks. The receive call is blocking and doesnt spam the thread.

LeeVangraefschepe committed:
Changed error to warning when connection fails

LeeVangraefschepe committed:
Set data now doenst get set when the queue is empty

AtlantiaKing commented on Networking/Leap/LeapClient.cpp:
```diff
@@ -158,6 +167,27 @@ bool leap::networking::LeapClient::HandleReceive()
+void leap::networking::LeapClient::HandleReceiveUDP()
+{
+    // Receive a response from the server
+    std::vector<char> buffer{};
```
This seems very unefficient because you create a new vector of, in this case, 1024 bytes everytime a new message is being received. 1kB per message on UDP seems a lot of allocations/deallocations. Maybe move the buffer to the UDPRun function and give a reference to the HandleReceiveUDP function?

LeeVangraefschepe replied:
This vector is moved to the packet. But there is an option to pool these buffers to decrease the usage of allocations & deallocations.

LeeVangraefschepe committed:
Removed return state when there is no connection

LeeVangraefschepe committed:
Reworked the way connecting works

The client code now reuses the buffer for the incoming packets. (copy is still made for the package itself)

LeeVangraefschepe committed:
Merge branch 'main' into Networking-fixes

LeeVangraefschepe committed:
Fixed issue where buffer was invalid (reverted back to old system)

Error messages are made more clear
